tree-sitter-go/tree-sitter-go.cabal

```diff
@@ -11,20 +11,38 @@ copyright:           2017 GitHub
 category:            Tree-sitter, Go
 build-type:          Simple
 
+common common
+  default-language: Haskell2010
+  ghc-options:
+    -Weverything
+    -Wno-all-missed-specialisations
+    -Wno-implicit-prelude
+    -Wno-missed-specialisations
+    -Wno-missing-import-lists
+    -Wno-missing-local-signatures
+    -Wno-monomorphism-restriction
+    -Wno-name-shadowing
+    -Wno-safe
+    -Wno-unsafe
+  if (impl(ghc >= 8.6))
+    ghc-options: -Wno-star-is-type
+  if (impl(ghc >= 8.8))
+    ghc-options: -Wno-missing-deriving-strategies
```

Comment:
We enable these warnings in every package and `import: common` in every component.